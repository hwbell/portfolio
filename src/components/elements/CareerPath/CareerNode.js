import React from "react";

import { toolsImages } from "../../../assets/toolsImages";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";

class CareerNode extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  renderTools() {
    let imageId;
    return this.props.tools.map((tool, i) => {
      imageId = `${this.props.job.split(" ").join("")}-tool${i}`;

      return (
        <div
          id={imageId}
          className="tool-image col"
          key={i}
          style={styles.toolHolder}
        >
          <a
            target="_blank"
            href={`https://www.google.com/search?q=${tool}`}
            alt={`link to ${tool}`}
          >
            <img
              // onMouseOver={() => {
              //   this.togglePopover(tool);
              // }}
              // onMouseLeave={() => {
              //   this.togglePopover();
              // }}
              style={styles.toolImage}
              src={toolsImages[tool]}
              alt="no image"
            />
          </a>
        </div>
      );
    });
  }

  render() {
    console.log(this.props.logo);
    const lineStyle = {
      height: "100px",
      borderLeft: "4px solid black",
    };
    return (
      <div className="career-node-holder col">
        <div className="career-node center-all-col">
          <p>{this.props.job}</p>

          {/* <div className="left-all-row row">{this.renderTools()}</div> */}
        </div>
        <div style={lineStyle}></div>
        <div className="left-all-row">
          <div className="year-block">{this.props.year}</div>
        </div>
      </div>
    );
  }
}

const styles = {
  toolHolder: {
    maxWidth: "50px",
    backgroundColor: "white",
  },
  toolImage: {
    height: "35px",
    width: "35px",
  },
  logo: {
    borderRadius: "10px",
    height: "100px",
  },
};

export default CareerNode;
