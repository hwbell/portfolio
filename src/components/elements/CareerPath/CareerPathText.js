const CareerPathText = [
  {
    year: "2010",
    job: "Research Associate at Brandeis University",
    tools: [ "matlab", "dna", "microscope"],
    logo: require("../../../assets/icons/brandeis.png")
  },
  {
    year: "2017",
    job: "Research Associate at University of Colorado",
    tools: [ "matlab", "dna", "microscope"],
    logo: require("../../../assets/icons/ucolorado.png")
  },
  {
    year: "2019",
    job: "Web Designer / Developer at Granicus Inc.",
    tools: [ "html5", "css3", "javascript", "php", "jira"],
    logo: require("../../../assets/icons/granicus.png")
  },
  {
    year: "2020",
    job: "The Next Step",
    tools: [ "html5", "css3", "javascript", "reactjs", "nodejs"],
    logo: require("../../../assets/icons/nextstep.png")
  }
]

export {
  CareerPathText
}