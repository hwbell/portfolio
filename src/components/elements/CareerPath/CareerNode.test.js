import React from 'react';
import ReactDOM from 'react-dom';
import CareerNode from './CareerNode';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { CareerPathText } from './CareerPathText';
// we'll use this obj as testing - CareerPathText[0]
// {
// year: "2010",
// job: "Brandeis University",
// tools: [ "matlab", "dna", "microscope"],
// logo: "./assets/someimage"
// }
// react router context
// import { MemoryRouter } from 'react-router-dom';

describe('CareerNode Component', () => {

  let wrapper;
  let someProps;
  beforeEach(() => {
    someProps = CareerPathText[0];

    wrapper = mount(
      // only need the MemoryRouter if there are links to routes of the react-router in the component that we need to test
      // <MemoryRouter>
      <CareerNode {...someProps} />
      // </MemoryRouter>
    );

  })

  // tests
  it('renders without crashing', async () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<CareerNode {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements for different data', () => {
    console.log(wrapper)

    expect(wrapper.find('.career-node').length).toEqual(1);

    // the tools will only render with the open modal
    // // there are 3 tools. there should be 3 tool-image elements
    // expect(wrapper.find('.tool-image').length).toEqual(3);

    // wrapper.setProps({tools: ["matlab", "dna"]})
    // // now there are 2 tools. there should be 2 tool-image elements
    // expect(wrapper.find('.tool-image').length).toEqual(2);

    // wrapper.setProps({tools: ["matlab", "dna", "backend", "jira", "protein chemistry", "databases"]})
    // // now there are 6 tools. there should be 6 tool-image elements
    // expect(wrapper.find('.tool-image').length).toEqual(6);

    // // are the right elements showing the right data?
    // expect(wrapper.find('.tool-image').at(0).props().id).toEqual('BrandeisUniversity-tool0')
    // expect(wrapper.find('.tool-image').at(5).props().id).toEqual('BrandeisUniversity-tool5')

  });

  it('receives the props passes to it', () => {
    expect(wrapper.props().year).toEqual('2010');
    expect(wrapper.props().job).toEqual('Research Associate at Brandeis University');
    expect(wrapper.props().tools).toEqual(["matlab", "dna", "microscope"]);
    expect(wrapper.props().logo).toEqual("brandeis.png");
  });

  // it('should render all buttons with info from provided props.buttons', () => {
  //   expect(wrapper.find('Link').at(0).props().to).toBe(someProps.buttons[0].link)
  //   expect(wrapper.find('Link').at(1).props().to).toBe(someProps.buttons[1].link)
  //   expect(wrapper.find('Link').at(2).props().to).toBe(someProps.buttons[2].link)

  // })
})