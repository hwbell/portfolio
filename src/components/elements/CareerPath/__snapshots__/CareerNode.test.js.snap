// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareerNode Component renders correctly 1`] = `
<div
  className="career-node-holder col"
>
  <div
    className="career-node center-all-col"
  >
    <p>
      Research Associate at Brandeis University
    </p>
  </div>
  <div
    style={
      Object {
        "borderLeft": "4px solid black",
        "height": "100px",
      }
    }
  />
  <div
    className="left-all-row"
  >
    <div
      className="year-block"
    >
      2010
    </div>
  </div>
</div>
`;
